### ISSUE TITLE
[FEATURE] NAT Helper Ping

### ISSUE THREAD

**Issue by volga629-1** [OpenSIPS/opensips#2165]
Opensips 3.1-dev 

Hello Everyone,
Based on this conversation 

https://groups.google.com/forum/#!topic/jssip/1D3n2_MJrf8

Opensips is not set user name portion in request uri for NAT helper OPTIONS packet. I found RFC for proxy layer that user name portion is not required, but I can't find any information about UA.

Is possible clarify what behavior should be for UA.

RFC

```
  The target of the OPTIONS request is identified by the Request-URI,
   which could identify another UA or a SIP server.  If the OPTIONS is
   addressed to a proxy server, the Request-URI is set without a user
   part, similar to the way a Request-URI is set for a REGISTER request.
```

![keepalive](https://user-images.githubusercontent.com/59034879/86942070-16dbc200-c11b-11ea-9372-284bbcf06573.png)

**Comment by razvancrainea** [comment]
Hi, Volga!

The OPTIONS messages the nat pinger module is sending are only sent with the purpose of keeping the NAT pinhole up, they are not actually targeted OPTION messages to a user. That's why the user is missing.
Feel free to switch this to a feature request in case this is a behavior you need.

Best regards,
Răzvan

**Comment by volga629-1** [comment]
@razvancrainea  Thank you for reply. Yes, that feature request.
Idea to have mod parameter which will allow add username in case ping goes to UA.


Example:

```

modparam("nathelper", "ping_proxy_only", 1)

```

**Comment by digipigeon** [comment]
+1 on this feature request, it would be very useful when using sip.js with SIP Ping.

**Comment by bogdan-iancu** [comment]
@InnovateAsterisk , the nathelper module is performing OPTION;s based pinging to the registered contact, so there is no need to copy anything into the registrar module - the nathelper IS taking the destination to be pinged from the registrar module.

I think the discussion here is about adding a username part also to the RURI in the OPTION requests.

**Comment by InnovateAsterisk** [comment]
> I think the discussion here is about adding a username part also to the RURI in the OPTION requests.

@bogdan-iancu Isn't there a route option for this, that I can add my own $rU based on my own conditions, rather than making up some kind of hard rule here. I just checked "local_route", but the message doesn't appear there. I can't seem to find the OPTIONS messages (when generated from sipping) in any of the routes.

**Comment by palmtown** [comment]
+1 - At minimum, should have an option to include the user to allow for a correct response from the client to be sent.

**Comment by palmtown** [comment]
Hello Community,

Please note that in that section of the RFC 3261 at , https://www.rfc-editor.org/rfc/rfc3261#section-11.2 , it states:

> The response to an OPTIONS is constructed using the standard rules for a SIP response as discussed in Section 8.2.6. The response code chosen MUST be the same that would have been chosen had the request been an INVITE.

Therefore, given the absence of the correct RURI in the OPTIONS request, any client that is in compliance with RFC 3261 would respond with 404 or alike as it would for an incorrect INVITE.  Therefore, wouldn't OpenSIPS not be following RFC 3261 if it is constructing an OPTIONS request as it would an INVITE?

**Comment by Shkiperon** [comment]
Hello everyone. I looked at list of commits in OpenSIPS 3.4 (released few days ago) and found commit https://github.com/OpenSIPS/opensips/commit/dbc36057be709daaaec576e7d9ea501b609bc7d7 with fix.
I have tested this issue (I saw it many times, so I know how to reproduce it) on OpenSIPS 3.4 built from source (in build options I disabled HP_MALLOC because there was an error with incorrect code usage on compile stage) and looks like it really fixed.

Can someone make some tests to confirm too?
